pkg/liqo-controller-manager/namespaceMap-controller/update_namespaceoffloading_status.go

```diff
@@ -0,0 +1,150 @@
+			if condition == nil {
+				continue
+			}
+			if condition.Status == corev1.ConditionTrue {
+				ready++
+			} else {
+				notReady++
+			}
+		}
+
+		switch {
+		case ready == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
+		case notReady == 0:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.ReadyOffloadingPhaseType
+		default:
+			namespaceOffloading.Status.OffloadingPhase = offv1alpha1.SomeFailedOffloadingPhaseType
+		}
+	} else {
+		namespaceOffloading.Status.OffloadingPhase = offv1alpha1.InProgressOffloadingPhaseType
+	}
+
+	if err := r.Update(context.TODO(), namespaceOffloading); err != nil {
+		klog.Errorf("%s -> unable to update namespaceOffloading for the namespace '%s'",
+			err, namespaceOffloading.Namespace)
+		return err
+	}
+
+	return nil
+}
+
+// This function sets ready condition according to remote namespace status, and checks global offloading status
+// with checkNamespaceOffloadingStatus function.
+func (r *NamespaceMapReconciler) setReadyNamespaceCondition(localName, clusterID string, status error) error {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: localName,
+		Name:      liqoconst.DefaultNamespaceOffloadingName,
+	}, namespaceOffloading); err != nil {
+		klog.Errorf("%s -> unable to get namespaceOffloading for the namespace '%s'", err, localName)
+		return err
+	}
+
+	var remoteNamespaceCondition offv1alpha1.RemoteNamespaceCondition
+	remoteNamespaceCondition.Type = offv1alpha1.NamespaceReady
+
+	if status == nil {
+		remoteNamespaceCondition.Status = corev1.ConditionTrue
+		remoteNamespaceCondition.Reason = "RemoteNamespaceCreated"
+		remoteNamespaceCondition.Message = fmt.Sprintf("Namespace '%s' correctly offloaded on cluster '%s'",
+			localName, clusterID)
+	} else {
+		remoteNamespaceCondition.Status = corev1.ConditionFalse
+		remoteNamespaceCondition.Reason = "RemoteNamespaceNotCreated"
+		remoteNamespaceCondition.Message = fmt.Sprintf("%s", status)
+	}
+
+	_ = liqoutils.UpdateNamespaceOffloadingCondition(&namespaceOffloading.Status, &remoteNamespaceCondition, clusterID)
+
+	return r.checkNamespaceOffloadingStatus(namespaceOffloading)
+}
+
+// The remote Namespace is requested to be deleted so the Ready Condition becomes "false". At the moment a remote
+// namespace is requested to be deleted only when the corresponding NamespaceOffloading resource is deleted, so
+// the global offloading phase is set to "Terminating".
+func (r *NamespaceMapReconciler) setNotReadyNamespaceCondition(localName, clusterID string) error {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: localName,
+		Name:      liqoconst.DefaultNamespaceOffloadingName,
+	}, namespaceOffloading); err != nil {
+		klog.Errorf("%s -> unable to get namespaceOffloading for the namespace '%s'", err, localName)
```

Comment:
Not Found is a possible error here? Should be caught differently?

Reply:
If the namespace is deleted, you will get a not found, or am I missing something?

Reply:
yes you are right, thank you.